### ISSUE TITLE
[UI] Something weird is going on with the timestamp search filter

### ISSUE THREAD

**Issue by JoyfulGen** [DDMAL/Rodan#1126]
I'm trying to find a specific resource based on its timestamp and I'm running into difficulties. I think there's a bug somewhere, but it's also possible I'm misunderstanding something, in which case this is a documentation issue. Here's what's happening:

My latest resources were created today (2024-01-23) around 14:21. If I open the table filters and enter that date and 02:21:10 PM in the first "Created" search bar, I get no results at all. If I switch the PM for AM, I get results both before and after 14:21, even though that's in the afternoon:
<img width="916" alt="Search AM" src="https://github.com/DDMAL/Rodan/assets/83373378/b4b5343c-6c6b-42b6-9aad-083ddd769621">

I assume that the two "Created" search bars can be used to search for an interval of time. If I enter the same date and 02:21:16 AM in the second search bar, I get no results:
<img width="914" alt="Search AM AM" src="https://github.com/DDMAL/Rodan/assets/83373378/3c95756d-9f18-476d-93da-7e61026db48d">

In that same scenario, if I switch the AM for PM in the second search bar, I once again get results around 14:21, but a lot of them are outside of the given time interval:
<img width="912" alt="Search AM PM" src="https://github.com/DDMAL/Rodan/assets/83373378/1566743e-ef55-4e19-860d-41da1ba831dc">

As a related issue, I think that if the timestamp search engine used a 24-hour clock like the resource timestamps, that would make searching much easier.

This issue doesn't impede rodan function, but it makes it near-impossible to locate a resource.

**Comment by fujinaga** [comment]
@softcat477 Can you look into this?

**Comment by yinanazhou** [comment]
Hi @JoyfulGen,I was wondering whether the timestamp search should include the exact time. Would it make more sense to filter based on the date only? Including the exact time—down to the second—seems a bit too granular to me.

**Comment by JoyfulGen** [comment]
@yinanazhou I agree that the seconds are definitely too much! However, I'm not sure that just the date is specific enough; if I've run workflows all morning, there will be a lot of jobs to search through for that date. Could we keep, say, the hour? I think the ideal for me would be to a have a search that you can refine: so you can search only by date if you want to, but you can also add a specific hour if that's needed. Is that possible?
